src/events.c

```diff
@@ -49,6 +84,10 @@ static void do_global_init()
         init_bgkill_state();
         init_input_device_state();
 
+        if (init_event_monitor()) {
```

Comment:
It's better to use opposite here: `if (! init_event_monitor())`
It's because in C programs the return result is usually implied by boolean, which mean 0 = FALSE.
When a programmer sees do_global_init() returns something, he would expect to see TRUE or FALSE (1 or 0).
TRUE means OK, FALSE means FAIL or ERROR.

Reply:
Yes and no. Yes, boolean logic works that way, but it's very common for C functions to return `0` on no error.

Reply:
What I mean:
```
        if (! init_event_monitor()) {
            ULOG_WARN_F("Failed to init event monitor");
        }
```
Instead of
```
        if (init_event_monitor()) {
            ULOG_WARN_F("Failed to init event monitor");
        }
```

Reply:
The exception to this rule is main() function which should return 0 on success.

Reply:
https://www.quora.com/What-is-the-difference-between-return-0-and-return-1

Reply:
Which function on your mind returns 0 on success?

Reply:
Random pick, read:

```
RETURN VALUE
       On success, the number of bytes read is returned (zero indicates end  of  file),
       and  the  file  position is advanced by this number.  It is not an error if this
       number is smaller than the number of bytes requested; this may happen for  exam‐
       ple  because fewer bytes are actually available right now (maybe because we were
       close to end-of-file, or because we are reading from a pipe, or  from  a  termi‐
       nal), or because read() was interrupted by a signal.  See also NOTES.

       On  error,  -1 is returned, and errno is set appropriately.  In this case, it is
       left unspecified whether the file position (if any) changes.
```

stat:

```
RETURN VALUE
       On success, zero is returned.  On error, -1 is returned, and errno is set appro‐
       priately.
```

This is also true for most linux system calls, glibc calls, etc.

Reply:
Which function is that?
I know some low level functions can return 0 on success. But it's not our case here.

Reply:
Also: `chown`, and also:

```
Originally the return value would have corresponded to an error code. Imagine there's a big table of all the possible erroneous outcomes a program may have. The program terminates; you want to know what, if anything, went wrong. If a non-zero integer were returned, you'd look it up in your table of error codes. If 0 is returned, it's equivalent to "nothing went wrong."
```

Also:

```
This is a remnant of the C/Unix legacy.

If your program is being called within a script, is is useful for the script to know whether your program was successful. Thus, was born the convention of returning 0 for success from executable programs.

C did not have native boolean types, so this was an integer. And this allowed us to return different numbers for different kinds of errors.

This convention then spread to regular functions, since, as others have posted, C/C++ did not originally have exceptions, and even once we had exceptions, it took discipline to use them correctly, and there was a lot of legacy code (and legacy programmers) using the return value convention.
```

https://stackoverflow.com/questions/2370986/what-does-returning-zero-by-convention-mean

Reply:
If you are talking about read() function, it can't return 1 on success, because it returns the number of bytes read.

Reply:
Just go through `man 2 syscall` and you will see many of them return 0 on success. I'm happy to change other parts of the code, but I'm not going to change it because you happen to not agree with my pick for return value. The function is only called once, it's an init function. ;)

Reply:
Your function doesn't return -1 on error and it doesn't set errno. It only returns two values, 0 on success, 1 on errror. Do you know other functions which do the same?

Reply:
Anyway, it's your code, it's your choice. I just give you my feedback.

Reply:
Alright. I'll leave it like this. If the function ever needs to init more then there are other error codes besides `1` that can be returned. Thanks for the review!